source/crud.txt

```diff
@@ -0,0 +1,118 @@
+=====================
+CRUD Operations Guide
+=====================
+
+.. default-domain:: mongodb
+
+.. toctree::
+   :caption: CRUD Operations
+   :hidden:
+
+   /crud/read-operations
+   /crud/write-operations
+   /crud/query-document
+   /crud/returned-data
+   /crud/compound-operations
+
+Overview
+--------
+
+CRUD (Create, Read, Update, Delete) operations allow you to work with
+the data stored inside of MongoDB. CRUD operations fall into two
+categories: read operations and write operations. All CRUD operations
+offer customizable options to tailor behavior for your application's
+specific needs. CRUD operations are all asynchronous.
+
+Read Operations
+---------------
+
+Read operations find and return documents stored within MongoDB.
+There are multiple types of read operation that allow you to
+access data in many different ways. While some read operations focus on
+allowing you to view data that exists in the database, other operations
+let you monitor the database for new data that matches certain criteria,
+and still other read operations let you apply a series of
+transformations and filters to all documents in a collection before
+returning a result set. The driver offers methods for each variant of
+read operation:
+
+Find
+~~~~
+
+If you want to view documents already in a collection, you can use
+:doc:`find() </usage-examples/find>` and
+:doc:`findOne() </usage-examples/findOne>`. These methods accept a
+query document that describes the documents you would like to view.
+While ``findOne()`` returns a single document, ``find()`` returns a
+:doc:`cursor </crud/read-operations/cursor>` that you can use to
+navigate matched documents.
+
+Aggregate
+~~~~~~~~~
+
+If you want to create custom processing pipelines for documents in a
+collection, you can use ``aggregate()``. This method accepts a
+pipeline of aggregation commands that are run in sequence. These
+commands let you filter, summarize, and augment documents in a
+collection into a result set for viewing.
+
+Watch / Subscribe
+~~~~~~~~~~~~~~~~~
+
+If you want to monitor a collection for new, update, replace, and
+deleted documents that match a certain criteria, you can use
+``watch()``. This method takes a pipeline of aggregation commands that
+are run in sequence on new data whenever write operations are run on
+the collection.
```

Comment:
Can we add an example here?

e.g. A pizza restaurant wants to get a notification whenever a new pizza order comes in